src/proc/color-formats-converter.cpp

```diff
@@ -354,6 +358,207 @@ namespace librealsense
+    // It is expected that all branching outside of the loop control variable will be removed due to constant-folding.
+    // This NV12i format is an INTERNAL FORMAT. Its configuration is as following: 2 lines of Y then one line of UV (line size is width)
+    // There is one Y value for each pixel, and one pair of U,V values for 4 pixels. 
+    // Their matching is similar to NV12 format - see: https://www.kernel.org/doc/html/v4.9/media/uapi/v4l/pixfmt-nv12.html?highlight=nv12
+    // For example: for the first 3 lines of the frame:
+    // Y0  Y1   Y2   Y3   ....
+    // Yw  Yw+1 Yw+2 Yw+3 .... (Yw:Ywidth)
+    // U0  V0   U1   V1
+    // The first pixel is (Y0, U0, V0), second pixel is (Y1, U0, V0)
+    // The first pixel in the second line is (Yw, U0, V0) second pixel in second line is (Yw+1, U0, V0)
+    // The third pixel in second line is (Yw+2, U1, V1)
+    template<rs2_format FORMAT> void unpack_nv12i(byte* const d[], const byte* s, int width, int height, int actual_size)
+    {
+        auto n = width * height;
+        assert(n % 16 == 0); // All currently supported color resolutions are multiples of 16 pixels. Could easily extend support to other resolutions by copying final n<16 pixels into a zero-padded buffer and recursively calling self for final iteration.
+
+        auto src = reinterpret_cast<const uint8_t*>(s);
+        auto dst = reinterpret_cast<uint8_t*>(d[0]);
+
+        auto src_height = height * 12 >> 3;
+
+        if (FORMAT == RS2_FORMAT_Y8)
+        {
+            // TODO REMI - check height is correct here (shall be equal to 1.5 * height - for height = 480, shall be 720)
+            for (int k = 0; k < src_height; k += 3)
+            {
+                // fill the y_buffer and uv_buffer
+                // while y is on 2 lines, and uv on the third line
+                auto start_of_y = src + k * width;
+                auto end_of_y = start_of_y + 2 * width;
+
+                std::vector<uint8_t> y_buffer(start_of_y, end_of_y);
+                librealsense::copy(dst, y_buffer.data(), 2 * width);
+                dst += 2 * width;
+            }
+            return;
+        }
+        if (FORMAT == RS2_FORMAT_Y16)
+        {
+            for (int k = 0; k < src_height; k += 3)
+            {
+                // fill the y_buffer and uv_buffer
+                // while y is on 2 lines, and uv on the third line
+                auto start_of_y = src + k * width;
+                auto end_of_y = start_of_y + 2 * width;
+                auto start_of_uv = end_of_y;
+                auto end_of_uv = start_of_uv + width;
+
+                std::vector<uint8_t> y_buffer(start_of_y, end_of_y);
+
+                for (int pix = 0; pix < 2 * width; pix += 16)
+                {
+                    uint16_t y[16];
+                    for (int i = 0; i < 16; ++i)
+                    {
+                        y[i] = y_buffer[pix + i];
+                    }
+                    librealsense::copy(dst, &y[0], 16);
```

Comment:
I think the bug could be here.. copy 16 bytes only when we build  uint16 array

Reply:
might want to do the same as yuv.
build 32 byte pairs of {0,val} for each 16 values and copy this 32 bytes to dst.
Just need to make sure dst can handle WxHx2Bpp

Reply:
Done